### ISSUE TITLE
Add `truncate`, `prefix`, and `suffix` params to `MaxLength`

### ISSUE THREAD

**Issue by todofixthis** [todofixthis/filters#15]
Adds truncating for too-long values, same as for `MaxBytes`, except that the result is truncated to a number of characters rather than a number of bytes.

**Comment by todofixthis** [comment]
Need to figure out how this is going to apply to non-string values..  Maybe we need a separate filter that is specifically for strings.. maybe like a `MaxChars` or similar 🤔

**Comment by todofixthis** [comment]
I like the idea of `MaxChars`.  I'll make its interface match `MaxBytes` as closely as possible:

```py
def __init__(
            self,
            max_chars: int,
            truncate: bool = False,
            prefix: typing.Text = '',
            suffix: typing.Text = '',
    ) -> None:
```

Note: no `encoding` param; this filter is designed to work exclusively with unicode strings (i.e., chain with `f.Unicode` if you need to decode a byte sequence).

Test cases:

- [x] `test_pass_none`
- [x] `test_pass_string_short`
- [x] `test_pass_string_short_with_prefix`
- [x] `test_pass_string_short_with_suffix`
- [x] `test_fail_string_too_long`
- [x] `test_pass_string_truncated`
- [x] `test_pass_string_truncated_with_prefix`
- [x] `test_pass_string_truncated_with_suffix`
- [x] `test_fail_wrong_type`
- [x] `test_error_max_chars_too_small`

**Comment by todofixthis** [comment]
Maybe for good measure I'll also add `truncate` to `MaxLength` (though I'm not really sure if `prefix` or `suffix` makes sense for that filter 😇).

**Comment by todofixthis** [comment]
Included in next release: 3.3.0
